lib/CXGN/Stock/Search.pm

```diff
@@ -274,12 +274,18 @@ sub search {
         $end = '';
     }
 
+    my $stock_synonym_cvterm_id = SGN::Model::Cvterm->get_cvterm_row($schema, 'stock_synonym', 'stock_property')->cvterm_id();
+    
     if ($any_name) {
-        $or_conditions = [
-            { 'me.name'          => {'ilike' => $start.$any_name.$end} },
-            { 'me.uniquename'    => {'ilike' => $start.$any_name.$end} },
-            { 'me.description'   => {'ilike' => $start.$any_name.$end} },
-        ];
+	$or_conditions = [
+	    { 'me.name'          => {'ilike' => $start.$any_name.$end} },
+	    { 'me.uniquename'    => {'ilike' => $start.$any_name.$end} },
+	    { 'me.description'   => {'ilike' => $start.$any_name.$end} },
+	    { -and => [
```

Comment:
the '-and' is for adding an 'or' condition: 

'name ilke $name OR uniquename ilike $name OR desc ilike $name 
OR 
(stockprop.value ILIKE $name AND stockprop.type_id = $cvterm_id_for_stock_synonym)

the syntax is weird in dbixClass